init test task

SpivakKonstantin opened the pull request:
Hello! Please check my test work.

jagermesh commented:
1) Docker version was conflicting with my local MariaDB
~~~
ERROR: for mariadb  Cannot start service mariadb: driver failed programming external connectivity on endpoint edoctrinatesttask_mariadb_1 (1829584d0623f9f935c5450e8aaa278bef513c33bfe78f9953c806fa133eb6a2): Error starting userland proxy: Bind for 0.0.0.0:3306 failed: port is already allocated
~~~
and apache
~~~
ERROR: for web  Cannot start service web: driver failed programming external connectivity on endpoint edoctrinatesttask_web_1 (20e2687539ef7ccc572b6a9ebcdc00cdcc0ffc23d5e0bd19f9817d055276f18f): Error starting userland proxy: Bind for 0.0.0.0:80: unexpected error (Failure EADDRINUSE)
~~~
So I have to stop my services. But this did not help, page showed 
~~~
502 Bad Gateway nginx/1.12.0
~~~

2) Regular version is generally ok, but has issues which we do not expect from Senior Developer

- JSHint errors in radio.js
~~~
radio.js: line 30, col 9, Function declarations should not be placed in blocks. Use a function expression or move the statement to the top of the outer function.
radio.js: line 39, col 9, Function declarations should not be placed in blocks. Use a function expression or move the statement to the top of the outer function.
radio.js: line 44, col 65, Missing semicolon.
radio.js: line 53, col 58, Missing semicolon.
radio.js: line 71, col 64, Missing semicolon.
radio.js: line 79, col 63, Missing semicolon.
radio.js: line 86, col 3, Missing semicolon.
~~~
- when you have two pages opened and finishing Quiz on one page - you can Save it on 2nd page without any errors. Score seems not updated though, which is good.
- some fields are named wrong, for example field to store score has name `count`. Could be typo, could be lack of attention or developer has issues with English
- lack in DB design - there is no table to store student answers - only result is stored
- node service listen on port 3000 which is usually blocked by firewalls
- some problems with instructions: there is no folder `app`, there is no requirement to make assets writable (`The directory is not writable by the Web process: /www/edoctrina-test-task/web/assets`)
- node service require `debug` package which was not in dependencies, actually there was no `package.json` at all, so first run throw error and I have to fix it by myself
~~~
module.js:487
    throw err;
    ^

Error: Cannot find module 'debug'
    at Function.Module._resolveFilename (module.js:485:15)
    at Function.Module._load (module.js:437:25)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/jagermesh/node_modules/engine.io/lib/transport.js:9:13)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
~~~
- other error was with bower configuration
~~~
The file or directory to be published does not exist: /www/edoctrina-test-task/vendor/bower/bootstrap/dist
~~~
all needed files were in `bower-assets` folder
![image](https://user-images.githubusercontent.com/1345694/27522504-a6007e58-59f4-11e7-968b-d69af5bf0e96.png)

SpivakKonstantin commented:
Hello! Thanks for your answer!
1) Docker:
You have ports conflict, I don't understand why didn't you stop your local service. I tested application only in  Docker environment, I wanted to show my knowledge of this technology. I've changed ports, so now you can start service locally on localhost:81

2)
Js - fixed

when you have two pages opened and finishing Quiz on one page - you can Save it on 2nd page without any errors. Score seems not updated though, which is good. 
------------------------
Yes, you can't save twice.

some fields are named wrong, for example field to store score has name count. Could be typo, could be lack of attention or developer has issues with English
------------------------
You are wrong. The field "count" contains the amount of questions. And field to store scores is named "result"

lack in DB design - there is no table to store student answers - only result is stored
------------------------
I didn't fiend this requirement in the test task.

node service listen on port 3000 which is usually blocked by firewalls
------------------------
I don't think that is a problem for test environment. But I fixed it to the same port(81)

some problems with instructions: there is no folder app, there is no requirement to make assets writable (The directory is not writable by the Web process: /www/edoctrina-test-task/web/assets)
------------------------
If you use Docker you won't have these problems

node service require debug package which was not in dependencies, actually there was no package.json at all, so first run throw error and I have to fix it by myself
------------------------
If you use Docker you won't have these problems

other error was with bower configuration
------------------------
If you use Docker you won't have these problems
